src/iosrv/vos.c

```diff
@@ -785,16 +786,17 @@ enum_unpack_recxs(daos_key_desc_t *kds, void *data,
 		return -DER_INVAL;
 
 	iod = &io->ui_iods[io->ui_iods_size - 1];
-	iod_akey = &iod->iod_name;
+	daos_iov_copy(&iod_akey, &iod->iod_name);
```

Comment:
Assume the memory allocation always succeeds?

Reply:
ah, ok.